### PR TITLE
Add pipelock test command for config validation

### PR DESCRIPTION
**Pull request by luckyPipewrench**
## Summary

- New `pipelock test` command with 30 built-in vectors across 9 scanner categories (DLP, blocklist, entropy, scheme, response injection, MCP response/input/tools, clean traffic)
- `--json` output and `--category` filtering for CI integration
- `--fail-on-gap` flag makes disabled security categories a hard failure (recommended for CI)
- Config-aware skip logic: vectors targeting disabled scanners show as `[SKIP]` with gap reporting
- Converts raw `os.Stderr` writes to cobra-idiomatic `cmd.PrintErrf()` across all CLI commands
- Fixes text report routing to stdout (cobra v1.10+ `cmd.Printf` defaults to stderr)
- Adds `pipelock test` step to CI workflow example

## Test plan

- [ ] `go test -race -count=1 ./internal/cli/... -run TestTest` — 22 test functions covering all modes
- [ ] `pipelock test` with default config — 23 pass, 7 skip
- [ ] `pipelock test --json` — valid JSON with correct counts
- [ ] `pipelock test --fail-on-gap` — exits 1 when scanners disabled
- [ ] `pipelock test --category dlp,entropy` — filtered run
- [ ] `pipelock test --category bogus` — rejects unknown categories
- [ ] Verify text output goes to stdout (`pipelock test > /tmp/out.txt` captures report)

## Summary by CodeRabbit

* **New Features**
  * Added a `test` command to validate security configurations with JSON output, category filtering, color control, and fail-on-gap behavior.

* **Improvements**
  * Standardized CLI output handling and improved exit code reporting.
  * CI now includes a "Test scanning coverage" workflow step.
  * Linter configuration refined to reduce false positives.

* **Tests**
  * Large, comprehensive test suite covering the new test command, reporting, and edge cases.

### CODE REVIEW

File: internal/cli/test.go [hunk 1]
```diff
@@ -0,0 +1,888 @@
...
+scheme enforcement, response injection, MCP scanning). Vectors that target
+disabled scanners are skipped, and reported as configuration gaps.
+
+Use --fail-on-gap in CI to ensure skipped security categories fail the
+gate. Without it, a config that disables most scanners can still exit 0.
+
+Exit codes:
+  0  All vectors passed (no gaps when --fail-on-gap is set)
+  1  One or more vectors failed, or gaps detected with --fail-on-gap
+  2  Config load error
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/cli/test.go:101]: Fixed in f74b533. Added `ExitError` type in `root.go` with `ExitCodeError()` and `ExitCodeOf()` helpers. Config load errors now wrap with exit code 2, and `main()` uses `ExitCodeOf(err)` instead of hardcoded `1`.

File: internal/cli/root.go [hunk 2]
```diff
@@ -2,9 +2,34 @@
...
+	Code int
+}
+
+func (e *ExitError) Error() string { return e.Err.Error() }
+func (e *ExitError) Unwrap() error { return e.Err }
+
+// ExitCodeError wraps err with a non-standard exit code.
+func ExitCodeError(code int, err error) error {
+	return &ExitError{Err: err, Code: code}
+}
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/cli/root.go]: Fixed in 9b02d5f. Added nil guard — `ExitCodeError(code, nil)` now returns nil instead of a panic-prone wrapper. Test added.

File: internal/cli/test.go [hunk 3]
```diff
@@ -0,0 +1,886 @@
...
+	header := fmt.Sprintf("Pipelock Test Suite — %d vectors, config: %s", report.Total, report.ConfigFile)
+
+	_, _ = fmt.Fprintln(out)
+	if color {
+		_, _ = fmt.Fprintf(out, "%s%s%s\n", ansiBold, header, ansiReset)
+		_, _ = fmt.Fprintf(out, "%s%s%s\n", ansiDim, strings.Repeat("\u2550", len(header)), ansiReset)
+	} else {
+		_, _ = fmt.Fprintln(out, header)
+		_, _ = fmt.Fprintln(out, strings.Repeat("=", len(header)))
+	}
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/cli/test.go:331]: Fixed in f5056a2 — both header and footer separators use `utf8.RuneCountInString` now.

File: internal/cli/root.go [hunk 4]
```diff
@@ -2,9 +2,38 @@
...
 )
 
+// ExitError wraps an error with a specific exit code for main() to use.
+type ExitError struct {
+	Err  error
+	Code int
+}
+
+func (e *ExitError) Error() string { return e.Err.Error() }
+func (e *ExitError) Unwrap() error { return e.Err }
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/cli/root.go:23]: Fixed in 382023a. `Error()` now returns `"exit code N"` when `Err` is nil, covering direct struct construction. Test added.
